assets/js/components/data-block.js

```diff
@@ -123,7 +119,7 @@ class DataBlock extends Component {
 						{ title }
 					</h3>
 					<div className="googlesitekit-data-block__datapoint">
-						{ `${ datapoint }${ datapointUnit }` }
+						{ datapointUnit ? numberFormatWithUnit( datapoint, datapointUnit ) : datapoint }
```

Comment:
See above, let's make this
```suggestion
						{ readableLargeNumber( datapoint, datapointUnit ) }
```

All usages of the `datapoint` prop should be updated to pass unformatted values. In cases where the current formatting (which should be removed) includes a unit, that unit should be passed as `datapointUnit` prop.